app/services/messaging/channel.rb

```diff
@@ -0,0 +1,35 @@
+class Messaging::Channel
+# On Error Handling: Any messaging channel implementation should raise
+# errors from the channel's API as exceptions. This is to allow background jobs to
+# retry in case of network/limit errors. The error object should contains the reason of
+# failure if it was due to a known error so that users of the service can use it
+# to handle known errors properly.
+
+  def initialize
+    @metrics = Metrics.with_object(self)
+  end
+
+  attr_reader :metrics
+
+  def send_message(recipient_number:, message:)
+    # To be implemented by the individual channel.
+  end
```

Comment:
This is the "common interface" for all sorts of channels.

Reply:
Addressed in [d583af2](https://github.com/simpledotorg/simple-server/pull/3589/commits/d583af25b720a8b9dcade4b5e0c92c3474ae3189)